extensions/default/src/ViewerLayout/index.jsx

```diff
@@ -86,11 +92,10 @@ function ViewerLayout({
 
             if (!isNested) {
               const { id, Component, componentProps } = toolDef;
-
               return <Component key={id} id={id} {...componentProps} />;
             } else {
               return (
-                <NestedMenu>
+                <NestedMenu icon={activeTool.icon} label={activeTool.label}>
```

Comment:
I think these props need to be exposed, but I only "radioGroup" tools in this menu would update the icon, if one is set? You might be able to determine this by. This is super tricky to "get right" in a reusable way that doesn't introduce coupling. We may need to do a "good enough" solution here, and create a follow-up issue.

You could bind your `setActiveTool` setState functions to:

- Tools that are in nested menus + of type `ohif.radioGroup`; when clicked, setActiveTool
- Tools that are not in a nested menu + of type `ohif.radioGroup`; when clicked, setActiveTool(defaulTool)

Long-term, you'd almost have to have the ToolbarService pub/sub tool config changes, or find a different/better way to expose toolbar "state" like this, and then just watch it.

Reply:
@dannyrb updated, lets discuss this new approach.